builtin_op_importers.cpp

```diff
@@ -1518,6 +1528,111 @@ DEFINE_BUILTIN_OP_IMPORTER(Size) {
   return {{weights}};
 }
 
+DEFINE_BUILTIN_OP_IMPORTER(Slice) {
+  ASSERT(inputs.size() == 1, ErrorCode::kINVALID_NODE);
+  nvinfer1::ITensor* tensor_ptr = &convertToTensor(inputs.at(0), ctx);
+  nvinfer1::Dims dims = tensor_ptr->getDimensions();
+  int nbDims = dims.nbDims;
+  OnnxAttrs attrs(node);
+  auto axes = attrs.get<std::vector<int>>("axes");
+  auto starts = attrs.get<std::vector<int>>("starts");
+  auto ends = attrs.get<std::vector<int>>("ends");
+  int H_idx = -1, W_idx = -1;
+  // This will be modified to contain expanded dims metadata,
+  // if needed (e.g. input tensor is 2D)
+  int artificial_dims = 0;
+  bool need_to_expand_dims = dims.nbDims < 3;
+
+  // Argument validation
+
+  // We don't support implicit indexing due to
+  // inability to deal with batch dim slicing
+  // (TRT doesn't support batch dim slicing)
+  ASSERT(axes.size() != 0, ErrorCode::kUNSUPPORTED_NODE);
+
+  // Since indexing is explicit, there must be
+  // equal number of axis, start and end indices
+  ASSERT((axes.size() == starts.size())
+    && (starts.size() == ends.size()), ErrorCode::kUNSUPPORTED_NODE);
+
+  for (size_t i = 0; i < axes.size(); ++i)
+  {
+    // We don't allow slicing batch dim, due to TRT limitations
+    ASSERT(axes[i] > 0 && axes[i] <= nbDims, ErrorCode::kUNSUPPORTED_NODE);
+
+    starts[i] = slice_clip_index(dims.d[axes[i]-1], starts[i]);
```

Comment:
[Copying comment from #82]
I'd prefer to remove the batch dim from axes/starts/ends in one place first, rather than deal with it every time they are used (i.e., having to add -1 everywhere etc.). But it's not a big deal and can be left as-is for now if you like.

Reply:
This can be simplified with the convert_axis utility function as per https://github.com/onnx/onnx-tensorrt/pull/103